rudder-web/src/main/scala/com/normation/rudder/web/services/ComplianceData.scala

```diff
@@ -95,6 +95,163 @@ object RulePolicyMode {
+        (Enforce, "<b>Enforce</b> is overriden by this <i><b>Node</b></i> mode")
+      case (None,Some(Enforce)) =>
+        (Enforce, "<b>Enforce</b> is overriden by this <i><b>Directive</b></i> mode")
+      case (Some(Audit), Some(Audit)) =>
+        (Audit, "<b>Audit</b> mode is overriden by both this <i><b>Node</b></i> and this <i><b>Directive</b></i>")
+      case (Some(Audit), None) =>
+        (Audit, "<b>Audit</b> mode is overriden by this <i><b>Node</b></i>")
+      case (Some(Audit), Some(Enforce)) =>
+        (Audit, "The <i><b>Directive</b></i> is configured to <b class='text-Enforce'>enforce</b> but is overriden to <b>audit</b> by this <i><b>Node</b></i>. ")
+      case (None,Some(Audit))  =>
+        (Audit, "<b>Audit</b> mode is overriden by this <i><b>Directive</b></i>")
+      case (Some(Enforce),Some(Audit))  =>
+        (Audit, "The <i><b>Node</b></i> is configured to <b class='text-Enforce'>enforce</b> but is overriden to <b>audit</b> by this <i><b>Directive</b></i>. ")
+    }
+  }
+
+  def ruleOnNode(nodeMode : Option[PolicyMode], globalMode : GlobalPolicyMode)( directivesMode : Set[Option[PolicyMode]]) : (String,String) = {
+   globalMode.overridable match {
+      case PolicyModeOverrides.Unoverridable =>
+        (globalMode.mode.name, s"""Rudder's global agent policy mode is set to <i><b>${globalMode.mode.name}</b></i> and is not overridable on a per-node or per-directive basis. Please check your Settings or contact your Rudder administrator.""")
+      case PolicyModeOverrides.Always =>
+        nodeMode match {
+          case Some(Audit) =>
+            (Audit.name,"<b>Audit</b> mode is overriden by this <i><b>Node</b></i>")
+          case nodeMode    =>
+
+            val (defaultMode,expl) = nodeMode match {
+              case Some(mode) =>
+                ( mode, s"<b>${mode.name}</b> mode is overriden by this <i><b>Node</b></i>")
+              case None =>
+                ( globalMode.mode, "This mode is the globally defined default. You can change it in the global <i><b>settings</b></i>.")
+            }
+            val default = (defaultMode.name,expl)
+
+            // We have a Set here so we only have 3 elem max (None, Some(audit), some(enforce))
+            directivesMode.toList match {
+              // No directives linked to the Rule, fallback to globalMode
+              case Nil => default
+              // We only have one element!!
+              case mode :: Nil => mode match {
+                case None => default
+                case Some(Audit) =>
+                  nodeMode match {
+                    case Some(Enforce) =>
+                     (Audit.name, "The <i><b>Node</b></i> is configured to <b class='text-Enforce'>enforce</b> but is overriden to <b>audit</b> by all <i><b>Directives</b></i>. ")
+                    case Some(Audit) =>
+                     (Audit.name, "<b>Audit</b> mode is overriden by both this <i><b>Node</b></i> and all <i><b>Directives</b></i>")
```

Comment:
And You're right about line 137, but i need to treat the case Some(Audit) since i have an Option[PolicyMode] (scala compiler / pattern matching does not authorize me here to have an Option[Enforce] ...) 

I wanted to add a comment here on my commenting phase ...